File: core/src/com/ore/infinium/WorldTime.java
package com.ore.infinium;


import java.time.LocalTime;

/**
 * ***************************************************************************
 * Copyright (C) 2014 by Shaun Reich <sreich02@gmail.com>                    *
 * *
 * This program is free software; you can redistribute it and/or            *
 * modify it under the terms of the GNU General Public License as           *
 * published by the Free Software Foundation; either version 2 of           *
 * the License, or (at your option) any later version.                      *
 * *
 * This program is distributed in the hope that it will be useful,          *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of           *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            *
 * GNU General Public License for more details.                             *
 * *
 * You should have received a copy of the GNU General Public License        *
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.    *
 * ***************************************************************************
 */
public class WorldTime {
    private LocalTime m_time;

    public void setTime(int hour, int minute, int second) {
        m_time = LocalTime.of(hour, minute, second);
    }

    public void tick(double elapsedTime) {

        /*
            uint8_t interval = 1;
    m_minute += interval;

    if (m_second >= 60) {
        ++m_minute;
        m_second -= 60;
    }

    if (m_minute >= 60) {
        ++m_hour;
        m_minute -= 60;
    }

    if (m_hour >= 24) {
        m_hour -= 24;
    }

    Q_ASSERT(m_minute <= 59 && m_second <= 59);
    Q_ASSERT(m_hour <= 23);
}

QString Time::toString() const
{
    QTime time(m_hour, m_minute, m_second);

    return time.toString("hh:mm:ss");
}
         */

    }

    public String timeString() {
        return m_time.toString();
    }
}

File: core/src/com/ore/infinium/WorldTime.kt
/**
MIT License

Copyright (c) 2016 Shaun Reich <sreich02@gmail.com>

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
 */

package com.ore.infinium

import java.time.LocalTime

class WorldTime {
    private var time: LocalTime? = null

    fun setTime(hour: Int, minute: Int, second: Int) {
        time = LocalTime.of(hour, minute, second)
    }

    fun tick(elapsedTime: Double) {

        /*
            uint8_t interval = 1;
    minute += interval;

    if (second >= 60) {
        ++minute;
        second -= 60;
    }

    if (minute >= 60) {
        ++hour;
        minute -= 60;
    }

    if (hour >= 24) {
        hour -= 24;
    }

    assert(minute <= 59 && second <= 59)
    assert(hour <= 23)
}

         */

    }

    fun timeString(): String {
        return time!!.toString()
    }
}

